src/Identity/IdentityServer/CustomTokenRequestValidator.cs

```diff
@@ -101,6 +109,20 @@ public async Task ValidateAsync(CustomTokenRequestValidationContext context)
             }
         }
 
+        // Attempts to find ssoConfigData for a given validate request subject
+        // this is actually guarenteed to pretty often be null, because more than just sso login requests will come
+        // through here
+        var ssoConfigData = await GetSsoConfigurationDataAsync(context.Result.ValidatedRequest.Subject);
+
+        // You can't put this below the user.MasterPassword != null check because TDE users can still have a MasterPassword
+        // It's worth noting that CurrentContext here will build a user in LaunchDarkly that is anonymous but DOES belong
+        // to an organization. So we will not be able to turn this feature on for only a single user, only for an entire 
+        // organization at a time.
+        if (ssoConfigData != null && _featureService.IsEnabled(FeatureFlagKeys.TrustedDeviceEncryption, CurrentContext))
+        {
+            context.Result.CustomResponse["MemberDecryptionOptions"] = CreateMemberDecryptionOptions(ssoConfigData, user).ToList();
```

Comment:
Might want to rename this to `UserDecryptionOptions` now

```suggestion
            context.Result.CustomResponse["UserDecryptionOptions"] = CreateUserDecryptionOptions(ssoConfigData, user).ToList();
```

Reply:
Good catch!